Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelDetectorObjectFactory.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelDetectorElement.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelToDetectorVolume.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelTree.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <GeoModelHelpers/getChildNodesWithTrf.h>
+
+#include "GeoModelKernel/GeoDefinitions.h"
+
+class GeoShape;
+struct GeoModelTree;
+class Surface;
+namespace Acts {
+class GeoModelDetectorObjectFactory {
+ public:
+  using GeoModelBoundingBox = std::shared_ptr<Experimental::DetectorVolume>;
+
+  struct Options {
+    std::vector<std::string> queries = {};
+  };
```

Comment:
I took that from the SurfaceFctory Dimitra did. I was also wondering why we need a struct for that. She said it is for some extensibility reason. Maybe you can add some information on that @dimitra97 ? Otherwise I am fine simplifying it.